cockpit/leapp.html

```diff
@@ -267,29 +294,142 @@
             output.empty();
             loader.hide();
             
-            show_target_ports();
+            release_lock_cmd();
+            check_target_ports();
+        }
+
+        function map_ports(ports_range, busy_ports, selected_ports) {
+
+            ports = {};
+            mapped_ports = {};
+
+            for (i in ports_range) {
+                ports[ports_range[i]] = [];
+            }
+
+            function get_available_port(min, max, excluded) {
+                if (max - min - excluded.length == 0)
+                    // if null is returned then docker should automatically assign port
+                    return null;
+                for (var port = min; port <= max; port++) {
+                    port = port.toString();
+                    if (excluded.indexOf(port) > -1)
+                        continue;
+                    return port;
+                }
+            }
+
+            for (i in selected_ports) {
+                port = selected_ports[i];
+                for (range in ports) {
+                    let interval = range.split('-');
+                    let min_p = parseInt(interval[0]);
+                    let max_p = parseInt(interval[1]);
+
+                    if (! (min_p <= port && port <= max_p)) {
+                        continue;
+                    }
+
+                    ports[range].push(port);
+                    if (port in busy_ports) {
+                        h_port = get_available_port(min_p, max_p, ports[range]);
+                    } else {
+                        h_port = port;
+                    }
+                    mapped_ports[h_port] = port;
+                }
+            }
+
+            return mapped_ports;
         }
 
-        function show_target_ports() {
-            // maybe it is a good place to unlock migrate button?
+        var ports_scan_range = [];
+        function check_target_ports() {
             if (! $("input[name=target-machine]:checked").length ||
                 ! $("input[name=source-machine]:checked").length) {
                 return;
             }
-            $("#target-ports ul li").remove();
+
+            var ip_addr = $("input[name=target-machine]:checked").data('ip');
+
+            var checked_ports_cnt = $("#panel-body input.source-ports:checked").length;
+            var min = null;
+            var max = null;
```

Comment:
can we call them min_port and max_port? min/max feels wrong especially since you use them as parameters in add_ports_range as well (Kind of confusing)